custom/conf/app.example.ini

```diff
@@ -957,6 +957,15 @@ LEVEL = Info
 
 ;; Allow fork repositories without maximum number limit
 ;ALLOW_FORK_WITHOUT_MAXIMUM_LIMIT = true
+;
+;; Enable applying a global size limit defined by REPO_SIZE_LIMIT. Each repository can have a value that overrides the global limit
+;; "false" means no limit will be enforced, even if specified on a repository
+;ENABLE_SIZE_LIMIT = false
```

Comment:
Why is this option even needed? I mean if by default global limit and individual repo limit is 0 (no limit). What is the use case for this option?

Reply:
@lafriks The reason why disabled by default is that it is a feature that should be enabled by administrator people mostly install gitea for their own need and such option might not be needed by default. It would be needed for any cloud install. So this allows gitea to work like before. 
Despite me believing it that feature is ready, I still believe it should be "experimental" hence disabled by default for some time.

Reply:
As for the use case it allows to easily switch on and off without modifying anything in the actual limits. Cause it could be quite hard after populating all the size limits individually on different repos to disable this feature. One would need to update each repo size limit (if he chooses not to use the global one).

Reply:
Yes but if you do not need limits you would not just set them, we have too many options already imho in app ini to add even more, as for being experimental we could state it in docs and UI in label that for example.

Reply:
I

> Main issue that still needs to be resolved is splitting repo size stored in database (git object size and LFS object size) in different columns

@lafriks is this 

> Of course refactor to split current repo size column into two can be made outside of this PR but this PR should depend on it being merged first

Ok from your point of view how this one is close to the needed? 
https://github.com/go-gitea/gitea/pull/22900

Reply:
yes that is exactly what is needed

Reply:
@lafriks is current apporach ok? Can I resolve this. if not - could you choose among the options or suggest something else?

Reply:
I also think we don't need this. We can use `REPO_SIZE_LIMIT = 0` to do the same thing.

Reply:
How about using the  REPO_SIZE_LIMIT = -1 as the switch off (and removing the boolean flag). 

REPO_SIZE_LIMIT = 0 Size limit is enabled but I don't care what the size limit is if an individual repo size limit is set use it.
REPO_SIZE_LIMIT > 0 Use this Size limit for any repository where size limit is 0 (not set) use individual repo size where it is set
Does this look acceptable? This would remove the need to for the extra config parameter also it would still allow ti switch off the size limit checking if administrator chooses to do so?